Lib/test/test_zipimport.py

```diff
@@ -718,6 +718,7 @@ def doTraceback(self, module):
             print_tb(tb, 1, s)
             self.assertTrue(s.getvalue().endswith(
                 '    def do_raise(): raise TypeError\n'
+                '' if sys.flags.no_debug_ranges else
```

Comment:
Not a big fan of this, there's several places in `test_traceback.py` we would have to do this. I wonder if there's a nicer way to achieve this...

Maybe some test helper that goes through and strips out lines with `*^^^^\n`?

Reply:
Managed to get around this by sticking a `@requires_ranges` on the test with a lot of carets and just handling the rest like this.